java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -132,6 +141,37 @@ public static Dataset commit(BufferAllocator allocator, String path,
   public static native Dataset commitAppend(String path, Optional<Long> readVersion,
       List<String> fragmentsMetadata);
 
+  /**
+   * Create a new Dataset Scanner.
+   *
+   * @param batchSize the scan options with batch size, columns filter, and substrait
+   * @return a dataset scanner
+   */
+  public Scanner newScan(long batchSize) {
+    return newScan(new ScanOptions(batchSize), Optional.empty());
+  }
```

Comment:
do we have `newScan()` with a default batchSize?

Reply:
change to allow Rust core to decide the default batch size, Lance Rust core has default batch size mechanism